std/range/package.d

```diff
@@ -1364,183 +1364,189 @@ Params:
 
 Returns:
     A range type dependent on $(D R1) and $(D R2).
-
-Bugs:
-    $(BUGZILLA 14660)
  */
 auto choose(R1, R2)(bool condition, R1 r1, R2 r2)
 if (isInputRange!(Unqual!R1) && isInputRange!(Unqual!R2) &&
     !is(CommonType!(ElementType!(Unqual!R1), ElementType!(Unqual!R2)) == void))
 {
-    static struct Result
+    return ChooseResult!(R1, R2)(condition, r1, r2);
+}
+
+private struct ChooseResult(R1, R2)
+{
+    import std.traits : hasElaborateCopyConstructor, hasElaborateDestructor;
+
+    private union
     {
-        import std.algorithm.comparison : max;
-        import std.algorithm.internal : addressOf;
-        import std.traits : hasElaborateCopyConstructor, hasElaborateDestructor;
+        R1 _r1;
+        R2 _r2;
+    }
+    private bool r1Chosen;
```

Comment:
Since you use the underscore prefix convention for the first two members, why not for this one as well?

Reply:
btw you may want to NOT use it here because it's a private data type anyway...

Reply:
good point. Have removed all the underscores.

Reply:
I got rid of the underscores.